source/class/qx/test/lang/Type.js

```diff
@@ -42,6 +42,156 @@ qx.Class.define("qx.test.lang.Type", {
+      var withInheritance = new Child();
+      this.assertFalse(isPojo(withInheritance));
+
+      //POJO with class prototype
+      var pojoWithClassPrototype = { age: 22 };
+      Object.setPrototypeOf(pojoWithClassPrototype, Person.prototype);
+      this.assertFalse(isPojo(pojoWithClassPrototype));
+
+      //POJO with an object prototype
+      //Should return true
+      var pojoWithObjectPrototype = { extraproperty: "Nuts" };
+      var proto = {
+        protoMethod() {
+          return "protomethod";
+        }
+      };
+      Object.setPrototypeOf(pojoWithObjectPrototype, proto);
+      this.assertTrue(isPojo(pojoWithObjectPrototype));
+
+      //Object which has a POJO prototype, which in turn has a class prototype
+      //Should return false
+      var prototypeIsObjectThenConstructor = { extraproperty: "Nuts" };
+      var proto = {
+        protoMethod() {
+          return "protomethod";
+        }
+      };
+
+      Object.setPrototypeOf(prototypeIsObjectThenConstructor, proto);
+      Object.setPrototypeOf(proto, Person.prototype);
+      this.assertFalse(isPojo(prototypeIsObjectThenConstructor));
+
+      //Object which has a POJO prototype, which in turn has a POJO prototype
+      //Should return true
+      var obj = {
+        nuts: 345
+      };
+
+      var proto1 = {
+        myMethod() {
+          return "proto1.myMethod";
+        }
+      };
+
+      Object.setPrototypeOf(obj, proto1);
+      var proto2 = {
+        myMethod() {
+          return "proto2.myMethod";
+        },
+
+        method2() {
+          return "proto2.method2";
+        }
+      };
+
+      Object.setPrototypeOf(proto1, proto2);
+
+      this.assertEquals("proto1.myMethod", obj.myMethod());
+      this.assertEquals("proto2.method2", obj.method2());
+      this.assertTrue(isPojo(obj));
+
+      //Object which is instance of a class, where the class has a prototype which is not Object.prototype
+      //Must return false
+      function NonObjectPrototype() {
+        this.foo = "foo";
+      }
+
+      var proto = {
+        myMethod() {
+          return "proto.myMethod";
+        }
+      };
+
+      NonObjectPrototype.prototype = proto;
+      NonObjectPrototype.prototype.constructor = NonObjectPrototype;
+      NonObjectPrototype.prototype.extraMethod = function () {
+        return "extraMethod";
+      };
+
+      var obj = new NonObjectPrototype();
+
+      this.assertEquals("proto.myMethod", obj.myMethod());
+      this.assertEquals("extraMethod", obj.extraMethod());
+      this.assertEquals("foo", obj.foo);
+      this.assertFalse(isPojo(obj));
+
+      //Qooxdoo objects must not be POJOs
+      var obj = new qx.core.Object();
+      this.assertFalse(isPojo(obj));
+
+      //ES6 class must not be a POJO
+      class ES6Class {
+        constructor() {
+          this.foo = "bar";
+        }
+
+        method() {
+          return "method";
+        }
+      }
+
+      var obj = new ES6Class();
+      this.assertFalse(isPojo(obj));
```

Comment:
Please add one more, as I don't think this is checked in other tests. (Feel free to rename variables to match the style of other tests you've implemented.)

```
const prototype = { num : 42 };
let x = { text : "hello", __proto__ : prototype };
this.assertTrue(isPojo(x);
```